src/iface/ethernet.rs

```diff
@@ -179,6 +179,11 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
                                         &eth_frame.src_addr());
                 }
 
+                for raw_socket in sockets.iter_mut().filter_map(Socket::as_raw) {
+                    let _ = raw_socket.process(timestamp, &IpRepr::Ipv4(ipv4_repr),
```

Comment:
You mean `return Err(e)`, not panic? Becuase you do `return Err(e)` for socket errors.